### DISCUSSION TITLE
Seems OCaml is implementing locality and borrowing features

### DISCUSSION THREAD

**Discussion by jkone27** [dotnet/fsharp#15521] [Ideas]
Is F# going to fetch similar features from base OCaml, it seems they would allow much better memory allocation and avoid GC completely in some programs. This would be a huge performance boost over C#???

https://blog.janestreet.com/oxidizing-ocaml-locality/

https://blog.janestreet.com/oxidizing-ocaml-ownership/

**Comment by vzarytovskii** [comment]
No immediate plans for any such features.
If you'd like to see it, it will need to be a language suggestion which will be reviewed by Don.
